src/api/app/instrumentations/workflow_instrumentation.rb

```diff
@@ -0,0 +1,21 @@
+module WorkflowInstrumentation
```

Comment:
What do you think of this approach to extract all instrumentation code to a separate file? Each model has its own instrumentation. The only change needed to use this instrumentation, beside including this module in the `Workflow` class, is to add `run_callbacks` to the methods for which we defined callbacks.